### DISCUSSION TITLE
firstOr... methods for simple collections

### DISCUSSION THREAD

**Discussion by hivokas** [laravel/framework#32888] [General]
What do you think about adding `firstOr...` (`firstOrFail`, `firstOr`, etc.) methods to simple collections?

So that we could:

```php
$collection->firstOr(function () {
    abort(403);
});
```

Instead of:

```php
$firstItem = $collection->first();

if (! $firstItem) {
    abort(403);
}
```

**Comment by netpok** [comment]
I generally don't like firstOr methods and and the abort helper but that's just my personal preference. The firstOrFail I don't know what should throw, maybe create an ElementNotFound exception.

The one liner for your specific use case is:
```php
$firstItem = $collection->first() ?? abort(403);
```

**Comment by royduin** [comment]
```
abort_if($collection->empty(), 403);
```
